functors task (2)/student/src/impl.cpp

```diff
@@ -7,48 +7,47 @@ void DataBrowser::userLeave(const std::string &userId)
 
 bool DataBrowser::getDataType1(const std::string &userId, std::vector<size_t> &returnValues) const
 {
-    const auto& it = m_dataReaders.find(userId);
-    if (it == m_dataReaders.cend())
+    auto function = []( IDataSelector* selector, std::vector<size_t>& returnValues ) -> bool
     {
-        return false;
-    }
-    if (it->second == nullptr)
-    {
-        return false;
-    }
-    return it->second->getDataType1(returnValues, 0);
+        return selector->getDataType1( returnValues, 0 );
+    };
+
+    return safeCall( userId, [this, &returnValues, function]( const std::unique_ptr<IDataSelector>& selector ) -> bool
```

Comment:
Hard to read. Consider naming this lambda.
```suggestion
    auto getter = [this, &returnValues, function]( const std::unique_ptr<IDataSelector>& selector ) ...
    return safeCall( userId, getter)
```